eclipse/src/saros/ui/wizards/AddContactWizard.java

```diff
@@ -136,133 +98,24 @@ public JID getContact() {
-                    return !DialogUtils.openQuestionMessageDialog(
-                        null, dialogContent.dialogTitle, dialogContent.dialogMessage);
-                  }
-                });
-      } catch (Exception e1) {
-        log.debug("Error opening questionMessageDialog", e);
-      }
-
-      if (cancel) throw new XMPPException(dialogContent.invocationTargetExceptionMessage);
-
-      log.warn(
-          "Problem while adding a contact. User decided to add contact anyway. Problem: "
-              + e.getMessage());
-    }
-
-    connection.getRoster().createEntry(jid.getBase(), nickname, null);
-  }
-
-  private DialogContent getDialogContent(XMPPException e) {
-
-    // FIXME: use e.getXMPPError().getCode(); !
-
-    if (e.getMessage().contains("item-not-found")) {
-      return new DialogContent(
-          "Contact Unknown",
-          "The contact is unknown to the XMPP server.\n\n"
-              + "Do you want to add the contact anyway?",
-          "Contact unknown to XMPP server.");
-    }
-
-    if (e.getMessage().contains("remote-server-not-found")) {
-      return new DialogContent(
-          "Server Not Found",
-          "The responsible XMPP server could not be found.\n\n"
-              + "Do you want to add the contact anyway?",
-          "Unable to find the responsible XMPP server.");
-    }
-
-    if (e.getMessage().contains("501")) {
-      return new DialogContent(
-          "Unsupported Contact Status Check",
-          "The responsible XMPP server does not support status requests.\n\n"
-              + "If the contact exists you can still successfully add him.\n\n"
-              + "Do you want to try to add the contact?",
-          "Contact status check unsupported by XMPP server.");
-    }
-
-    if (e.getMessage().contains("503")) {
-      return new DialogContent(
-          "Unknown Contact Status",
-          "For privacy reasons the XMPP server does not reply to status requests.\n\n"
-              + "If the contact exists you can still successfully add him.\n\n"
-              + "Do you want to try to add the contact?",
-          "Unable to check the contact status.");
-    }
-
-    if (e.getMessage().contains("No response from the server")) {
-      return new DialogContent(
-          "Server Not Responding",
-          "The responsible XMPP server is not connectable.\n"
-              + "The server is either inexistent or offline right now.\n\n"
-              + "Do you want to add the contact anyway?",
-          "The XMPP server did not respond.");
-    }
-
-    return new DialogContent(
-        "Unknown Error",
-        "An unknown error has occured:\n\n"
-            + e.getMessage()
-            + "\n\n"
-            + "Do you want to add the contact anyway?",
-        "Unknown error: " + e.getMessage());
+  private void addContact(JID jid, String nickname) throws SarosCancellationException {
+    BiPredicate<String, String> questionDialog =
```

Comment:
Very questionable usage of a BiPredicate as a callback function, just because it works does not mean that it should be used that way.